scripts/s_aureus/generate_results.py

```diff
@@ -123,93 +121,60 @@ def build_csv_list(sample):
-    print('sequencing_technology: ' + sequencing_technology)
-
-    # Make a list so we can iterate over this several times easily
-    # TODO change to single sample -- get sample another way
-    metadata = list(csv.DictReader(open(Path(metadata_file))))
-
-    sample = metadata[0]
+    # Get sample_id and genome_assembly_name
+    with open('genome_assembly_name.txt') as genome_assembly_name_file:
+        gaf = json.load(genome_assembly_name_file)
+        genome_assembly_name = gaf['genome_assembly_name']
+        sample_id = gaf['sample_id']
+    print('genome_assembly_name: ' + genome_assembly_name)
+    print('sample_id: ' + sample_id)
 
-    csv_list = build_csv_list(sample)
+    csv_list = build_csv_list(sample_id)
 
-    with open(output_csv_file, 'w', encoding='UTF8') as csv_out:
+    with open('{sample_id}-results.csv'.format(sample_id=sample_id), 'w', encoding='UTF8') as csv_out:
         writer = csv.writer(csv_out)
         writer.writerow([x.header for x in csv_list])
         writer.writerow([x.value for x in csv_list])
 
-    # TODDO make this explicitly single sample. I will collate the results in another step
     # Build resultfiles.json
     json_out = dict()
-    for sample in metadata:
-        out_files = list()
-        for of in OUTPUT_FILES:
-            # Make sure file and type not already in list (psga framework required no duplicates)
-            # TODO in production version error rather than fixing the problem with a warning
-            file_in_list = False
-            f = of.filepath.format(output_path=output_path, sample_id=sample['SAMPLE_ID'])
-            t = of.typedescription
-            for item in out_files:
-                if item["file"] == f or item["type"] == t:
-                    print('file: ', item["file"])
-                    print('type: ', item["type"])
-                    print('f: ', f)
-                    print('t: ', t)
-                    file_in_list = True
-            if not file_in_list:
-                out_files.append({
-                    "file": f,
-                    "type": t
-                })
-            else:
-                print('WARNING: found duplicate file {t} : {f}'.format(t=t, f=f))
-        json_out[sample['SAMPLE_ID']] = out_files
-
-    with open(output_json_file, 'w') as f:
+    out_files = list()
+    for of in OUTPUT_FILES:
+        # Make sure file and type not already in list (psga framework required no duplicates)
+        # TODO in production version error rather than fixing the problem with a warning
+        file_in_list = False
+        f = of.filepath.format(
+            output_path=output_path, sample_id=sample_id, genome_assembly_name=genome_assembly_name)
+        t = of.typedescription
+        for item in out_files:
+            if item["file"] == f or item["type"] == t:
+                print('file: ', item["file"])
+                print('type: ', item["type"])
+                print('f: ', f)
+                print('t: ', t)
```

Comment:
f-strings ?

Reply:
f is the variable that holds the filename
unrelated to f-strings

Reply:
What I meant was that the code above can / should be implemented using f-strings, not `f` as a variable (which should be an explicit name rather than a letter).